### PR TITLE
RHCLOUD-12591: Add getSystemProfileBatch() to inventory impl

### PR DESCRIPTION
**Pull request by aleccohan**
Added getSystemProfileBatch() to both inventory impl() as well as xjoin implementations.  This is going to be used for an additional check wrt adding cert auth to remediations /playbook endpoint.

## Secure Coding Practices Checklist GitHub Link
    - https://github.com/RedHatInsights/secure-coding-checklist

    ## Secure Coding Checklist
    - [ ] Input Validation
    - [ ] Output Encoding
    - [ ] Authentication and Password Management
    - [ ] Session Management
    - [ ] Access Control
    - [ ] Cryptographic Practices
    - [ ] Error Handling and Logging
    - [ ] Data Protection
    - [ ] Communication Security
    - [ ] System Configuration
    - [ ] Database Security
    - [ ] File Management
    - [ ] Memory Management
    - [ ] General Coding Practices

### CODE REVIEW

File: src/connectors/inventory/impl.js [hunk 1]
```diff
@@ -90,6 +91,57 @@ module.exports = new class extends Connector {
...
+        try {
+            response = await this.doHttp({
+                uri: uri.toString(),
+                method: 'GET',
+                json: true,
+                rejectUnauthorized: !insecure,
+                headers: this.getForwardedHeaders()
+            },
+            {
+                key: `remediations|http-cache|inventory|${ids.join()}`,
```
Comment (jharting) [inline src/connectors/inventory/impl.js:123]: this is the same key as in getSystemDetailsBatch above i.e. these two functions will keep overwriting the cache entry

File: src/connectors/inventory/impl.js [hunk 2]
```diff
@@ -90,6 +91,57 @@ module.exports = new class extends Connector {
...
+
+        if (ids.length > pageSize) {
+            const chunks = _.chunk(ids, pageSize);
+            const results = await P.map(chunks, chunk => this.getSystemProfileBatch(chunk, refresh));
+            return _.assign({}, ...results);
+        }
+
+        const uri = this.buildHostsUri();
+        uri.segment(ids.join());
+        uri.segment('system_profile');
```
Comment (jharting) [inline src/connectors/inventory/impl.js:109]: Like with the xjoin connector you can now limit the fields to only those you're interested in (owner id) using sparse fieldset.

**Review by jharting — CHANGES_REQUESTED** [review]

File: src/connectors/inventory/impl.js (hunk 2 above)
Comment (aleccohan) [reply src/connectors/inventory/impl.js:109]: ah!  that's great to know, I'll make that change.

**Comment by aleccohan** [comment]
@jharting new changes pushed up, when you have the chance let me know what you think!

**Review by jharting — APPROVED** [review]

**Pull request merged**